service/importer/definition.go

```diff
@@ -1,108 +1,111 @@
+	Name string `yaml:"name" json:"name,omitempty" validate:"printascii"`
 
 	// Description is the description of event.
-	Description string `yaml:"description" json:"description,omitempty"`
+	Description string `yaml:"description" json:"description,omitempty" validate:"printascii"`
 
 	// Data holds the input parameters of event.
-	Data map[string]*Parameter `yaml:"data" json:"data,omitempty"`
+	Data map[string]*Parameter `yaml:"data" json:"data,omitempty" validate:"required,dive,keys,printascii,endkeys,required"`
 }
 
 // Dependency represents a Docker container and it holds instructions about
 // how it should run.
 type Dependency struct {
 	// Image is the Docker image.
-	Image string `yaml:"image" json:"image,omitempty"`
+	Image string `yaml:"image" json:"image,omitempty" validate:"printascii"`
 
 	// Volumes are the Docker volumes.
-	Volumes []string `yaml:"volumes" json:"volumes,omitempty"`
+	Volumes []string `yaml:"volumes" json:"volumes,omitempty" validate:"unique,dive,printascii"`
 
 	// VolumesFrom are the docker volumes-from from.
-	VolumesFrom []string `yaml:"volumesfrom" json:"volumefrom,omitempty"`
+	VolumesFrom []string `yaml:"volumesfrom" json:"volumefrom,omitempty" validate:"unique,dive,printascii"`
 
 	// Ports holds ports configuration for container.
-	Ports []string `yaml:"ports" json:"ports,omitempty"`
+	Ports []string `yaml:"ports" json:"ports,omitempty" validate:"unique,dive,numeric"`
 
 	// Command is the Docker command which will be executed when container started.
-	Command string `yaml:"command" json:"command,omitempty"`
+	Command string `yaml:"command" json:"command,omitempty" validate:"printascii"`
 
 	// Args hold the args to pass to the Docker container
-	Args []string `yaml:"args" json:"args,omitempty"`
+	Args []string `yaml:"args" json:"args,omitempty" validate:"dive,printascii"`
 
 	// Env is a slice of environment variables in key=value format.
-	Env []string `yaml:"env" json:"env,omitempty"`
+	Env []string `yaml:"env" json:"env,omitempty" validate:"unique,dive,printascii"`
 }
 
 // Task describes a service task.
 type Task struct {
 	// Name is the name of task.
-	Name string `yaml:"name" json:"name,omitempty"`
+	Name string `yaml:"name" json:"name,omitempty" validate:"printascii"`
 
 	// Description is the description of task.
-	Description string `yaml:"description" json:"description,omitempty"`
+	Description string `yaml:"description" json:"description,omitempty" validate:"printascii"`
 
 	// Inputs are the definition of the execution inputs of task.
-	Inputs map[string]*Parameter `yaml:"inputs" json:"inputs,omitempty"`
+	Inputs map[string]*Parameter `yaml:"inputs" json:"inputs,omitempty" validate:"dive,keys,printascii,endkeys,required"`
 
 	// Outputs are the definition of the execution results of task.
-	Outputs map[string]*Output `yaml:"outputs" json:"outputs,omitempty"`
+	Outputs map[string]*Output `yaml:"outputs" json:"outputs,omitempty" validate:"required,dive,keys,printascii,endkeys,required"`
```

Comment:
```suggestion
	Outputs map[string]*Output `yaml:"outputs" json:"outputs,omitempty" validate:"required,dive,keys,printascii,endkeys"`
```

Reply:
Ince case when a task doesn't necessarily need to send an output then we might allow empty outputs. But on the other hand, sending outputs _always_ can be considered as a good practice because there is no another way of confirming if a task is successfully executed by the service or not.